layers/+lang/zig/config.el

```diff
@@ -0,0 +1,20 @@
+;;; config.el --- zig Layer config File for Spacemacs
+;;
+;; Copyright (c) 2012-2021 Sylvain Benner & Contributors
+;;
+;; Author: Riccardo Binetti <rbino@gmx.com>
+;; URL: https://github.com/syl20bnr/spacemacs
+;;
+;; This file is not part of GNU Emacs.
+;;
+;;; License: GPLv3
+
+(spacemacs|define-jump-handlers zig-mode)
+
+(defvar zig-backend nil
```

Comment:
The backend variable should support two values, of which one must be `lsp` the other one should be the simple choice for non LSP users. Nil is reserved for `not set` resulting in an automatic choice of backends.
In your case lets introduce `(lsp zig-mode)` and describe both alternative in the README.

`zls` would be a specific implementation of an LSP server for which there is a different standard variable, see the python layer for details.

However with only one server supported introducing the server choice variable is not necessary.